File: src/Test2.java
import jdk.nashorn.api.scripting.ScriptObjectMirror;

import javax.script.ScriptEngine;
import javax.script.ScriptEngineManager;
import java.io.File;
import java.io.FileReader;
import java.util.Arrays;

/**
 * js调用Java方法
 */
public class Test2 {


    public static void main(String[] args) throws Exception {
        String relativelyPath = System.getProperty("user.dir");
        System.out.println(relativelyPath);
        ScriptEngine engine = new ScriptEngineManager().getEngineByName("nashorn");
        engine.eval(new FileReader(new File(relativelyPath + "/src/js/test2.js")));
    }

    public static String fun(String name) {
        System.out.format("我是java, %s", name);
        return "从java获取的值";
    }

    public static void fun2(Object object) {
        System.out.println(object.getClass());
        System.out.println("===============================");
    }
    public static void fun3(ScriptObjectMirror mirror) {
        System.out.println(mirror.getClassName() + ": " + Arrays.toString(mirror.getOwnKeys(true)));
        String foo = (String) mirror.get("key1");
        String bar = (String) mirror.get("key2");

        System.out.println("===============================");
    }

    public static void fun4(ScriptObjectMirror person) {
        System.out.println("Full Name is: " + person.callMember("getFullName"));
    }
}

File: src/js/test2.js
var test2 = Java.type('Test2');
var result = test2.fun('樊启龙');
print('\n' + result);

test2.fun2(123);
test2.fun2(49.99);
test2.fun2(true);
test2.fun2("hi there");
test2.fun2(String("bam"));
test2.fun2(new Number(23));
test2.fun2(new Date());
test2.fun2(new RegExp());
test2.fun2({foo: 'bar'});

print('passing object hash:');
test2.fun3({
    key1: 'bar',
    key2: 'foo'
});

function Person(firstName, lastName) {
    this.firstName = firstName;
    this.lastName = lastName;
    this.getFullName = function() {
        return this.firstName + " " + this.lastName;
    }
}
var person1 = new Person("Peter", "Parker");
test2.fun3(person1);
test2.fun4(person1);